e2e/testcases/reconciler_manager_test.go

```diff
@@ -55,6 +63,149 @@ const (
+
+	for _, obj := range rootSyncFriends {
+		err := nt.Validate(obj.GetName(), obj.GetNamespace(), obj)
+		require.NoError(t, err)
+	}
+
+	t.Log("Validate the RepoSync")
+	repoSync := &v1beta1.RepoSync{}
+	setNN(repoSync, client.ObjectKey{Name: configsync.RepoSyncName, Namespace: testNamespace})
+	err = nt.Watcher.WatchObject(kinds.RepoSyncV1Beta1(), repoSync.Name, repoSync.Namespace, []testpredicates.Predicate{
+		testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus),
+		testpredicates.HasFinalizer(metadata.ReconcilerManagerFinalizer),
+	})
+	require.NoError(t, err)
+
+	t.Log("Validate the RepoSync reconciler and its dependencies")
+	var repoSyncFriends []client.Object
+
+	repoSyncReconciler := &appsv1.Deployment{}
+	setNN(repoSyncReconciler, core.NsReconcilerObjectKey(repoSync.Namespace, repoSync.Name))
+	repoSyncFriends = append(repoSyncFriends, repoSyncReconciler)
+
+	// Note: reconciler-manager no longer applies ConfigMaps to configure the
+	// reconciler. So we don't need to validate their deletion. The deletion
+	// only happens when upgrading from a very old unsupported version.
+
+	repoSyncRB := &rbacv1.RoleBinding{}
+	setNN(repoSyncRB, client.ObjectKey{
+		Name:      controllers.RepoSyncPermissionsName(),
+		Namespace: testNamespace,
+	})
+	repoSyncFriends = append(repoSyncFriends, repoSyncRB)
+
+	repoSyncSA := &corev1.ServiceAccount{}
+	setNN(repoSyncSA, client.ObjectKeyFromObject(repoSyncReconciler))
+	repoSyncFriends = append(repoSyncFriends, repoSyncSA)
+
+	// See nomostest.CreateNamespaceSecret for creation of user secrets.
+	// This is a managed secret with a derivative name.
+	repoSyncAuthSecret := &corev1.Secret{}
+	setNN(repoSyncAuthSecret, client.ObjectKey{
+		Name:      controllers.ReconcilerResourceName(repoSyncReconciler.Name, nomostest.NamespaceAuthSecretName),
+		Namespace: repoSyncReconciler.Namespace,
+	})
+	repoSyncFriends = append(repoSyncFriends, repoSyncAuthSecret)
+
+	// See nomostest.CreateNamespaceSecret for creation of user secrets.
+	// This is a managed secret with a derivative name.
+	// For local kind clusters, the CA Certs are provided to authenticate the git server.
+	if nt.GitProvider.Type() == e2e.Local {
+		repoSyncCACertSecret := &corev1.Secret{}
+		setNN(repoSyncCACertSecret, client.ObjectKey{
+			Name:      controllers.ReconcilerResourceName(repoSyncReconciler.Name, nomostest.NamespaceAuthSecretName),
+			Namespace: repoSyncReconciler.Namespace,
+		})
+		repoSyncFriends = append(repoSyncFriends, repoSyncCACertSecret)
+	}
+
+	for _, obj := range repoSyncFriends {
+		err := nt.Validate(obj.GetName(), obj.GetNamespace(), obj)
+		require.NoError(t, err)
+	}
+
+	t.Log("Delete the RootSync and wait for it to be not found")
+	err = nomostest.DeleteObjectsAndWait(nt, rootSync)
+	require.NoError(t, err)
+
+	t.Log("Validate the RootSync reconciler and its dependencies were deleted")
+	for _, obj := range rootSyncFriends {
+		gvk, err := kinds.Lookup(obj, nt.Scheme)
```

Comment:
WDYT about putting the body of this in a taskgroup so that we can validate the friends in parallel?

Reply:
It's just a bunch of Get calls which the apiserver caches and the QPS is pretty high. I don't think doing 3-5 in parallel would really be all that much of time savings unless they're timing out.

Reply:
ack, sgtm